### PR TITLE
fix(animations): retain triggers values for moved tracked list items

### LINKED ISSUE #29526: Angular animation: :leave animation starts at wrong position
### Affected Package
@angular/animations

### Description
:leave animation has incorrect initial position for the animated elements.

## 🔬 Minimal Reproduction
https://stackblitz.com/edit/angular-fqxnxd?file=src%2Fapp%2Fapp.component.ts

Follow  the steps in the demo. Notice that only first element exits correctly.

## 🌍  Your Environment
Chrome v73

**Angular Version:**
"@angular/animations": "^7.2.10",
"@angular/common": "7.2.5",
"@angular/compiler": "7.2.5",
"@angular/core": "7.2.5"

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
when reordering a list with animations the list items lose their current
triggers values, that is because the reordering of an item is implemented
as follows (_note:_ the following implementation has been added in PR #23534)
 - the item is removed and marked _setForRemoval_
 - the item is re-inserted, and the _setForRemoval_ is changed to _setForMove_
 - the player set for animating the removal is destroyed when _setForMove_ is detected

the above steps allow the element not to be animated and to keep its styling but the
triggers values get lost since the removal transition/player has already been initialized,
so this change adds a _previousTriggersValues_ map in the _REMOVAL_FLAG_ field in order to
restore the triggers values if/when the removal turns out to be a move, changing the above steps to:
 - the item is removed and marked setForRemoval __and its current triggers values are saved as well__
 - the item is re-inserted, and the setForRemoval is changed to setForMove
 - the player set for animating the removal is destroyed when setForMove is detected __and the
   trigger values are re-applied in the engine's statesByElement map__

resolves #29526

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Issue

Issue Number: #29526

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

___

@jessicajaniuk 😄

### CODE REVIEW

File: packages/animations/browser/src/render/transition_animation_engine.ts [hunk 1]
```diff
@@ -991,8 +1001,21 @@ export class TransitionAnimationEngine {
 
         if (this.collectedEnterElements.length) {
           const details = element[REMOVAL_FLAG] as ElementAnimationState;
-          // move animations are currently not supported...
+          // animations for move operations (elements being removed and reinserted,
+          // e.g. when the order of an *ngFor list changes) are currently not supported
```
Comment (dario-piotrowicz) [inline packages/animations/browser/src/render/transition_animation_engine.ts:1005]: the previous comment was really not that clear to me so I expanded upon it after understanding what it meant, I hope that is fine 🙂

Comment (jessicajaniuk) [reply packages/animations/browser/src/render/transition_animation_engine.ts:1005]: Thank you for expanding it to be more clear.

**Review by jessicajaniuk — APPROVED** [review]
LGTM :cookie:

This is probably the only way this issue can be resolved given the constraints. Thanks for looking into this, @dario-piotrowicz!

**Comment by atscott** [comment]
This PR was merged into the repository by commit 8ba43a1a1ea811ad25d6af4384e1cce5a53b0baf.

**Pull request closed without merging**